sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/AsyncDocumentClient.java

```diff
@@ -609,7 +610,7 @@ Mono<ResourceResponse<Document>> upsertDocument(String collectionLink, Object do
      * @param options      the request options.
      * @return a {@link Mono} containing the single resource response with the replaced document or an error.
      */
-    Mono<ResourceResponse<Document>> replaceDocument(String documentLink, Object document, RequestOptions options);
+    Mono<ResourceResponse<Document>> replaceDocument(String documentLink, Object document, RequestOptions options, String collectionLink);
```

Comment:
there is a utils method exists which can help extract the collectionLink from the documentLink, probably can just use that:

    `        String collectionLink = Utils.getCollectionName(documentLink);`

Reply:
Thanks! Reduces touched files - fixed in next iteration.